FF1Lib/ItemPlacement.cs

```diff
@@ -175,6 +193,47 @@ public List<IRewardSource> PlaceSaneItems(MT19337 rng)
 				treasurePool = treasurePool.Select(treasure => generator.GetItem(rng)).ToList();
 			}
 
+			if (_flags.BetterTrapChests)
+			{
+				// First we'll make a list of all 'notable' treasure.
+				var notableTreasureList = new List<Item>()
+					.Concat(ItemLists.UberTier)
+					.Concat(ItemLists.LegendaryWeaponTier)
+					.Concat(ItemLists.LegendaryArmorTier)
+					.Concat(ItemLists.RareWeaponTier)
+					.Concat(ItemLists.RareArmorTier);
+				// Convert the list to a HashSet since we'll be doing lookups in it.
+				var notableTreasure = new HashSet<Item>(notableTreasureList);
+
+				// We sort the treasure pool based on value (sort of) and pull out the highest ranked ones to put
+				// in the trap chests we picked out.
+				var notableTreasurePool = treasurePool.FindAll(item => notableTreasure.Contains(item));

```

Comment:
You might want .Where instead of .FindAll but I'm honestly not sure what the difference is. My hunch would be to use .Where and ToList() at the end, and then use spliceRandom() later instead of First() and Remove() (since it doesn't both sorta)